src/MIDebugEngine/Engine.Impl/ExceptionManager.cs

```diff
@@ -278,6 +278,28 @@ public void SetAllExceptions(enum_EXCEPTION_STATE dwState)
             }
         }
 
+        public int TryGetExceptionBreakpoint(string bkptno, out EXCEPTION_INFO exceptionInfo)
+        {
+            const string CppExceptionCategoryString = "{3A12D0B7-C26C-11D0-B442-00A0244A1DD2}";
+            exceptionInfo = new EXCEPTION_INFO();
+            ExceptionCategorySettings categorySettings;
+            if (_categoryMap.TryGetValue(new Guid(CppExceptionCategoryString), out categorySettings))
+            {
+                ulong breakpointNumber = Convert.ToUInt32(bkptno, CultureInfo.InvariantCulture);
+                if (categorySettings.CurrentRules.ContainsValue(breakpointNumber))
+                {
+                    // test -- need to delete; these are placeholder values
+                    exceptionInfo.bstrExceptionName = categorySettings.CategoryName;
```

Comment:
>categorySettings.CategoryName [](http://example.com/codeflow?start=54&length=29)

FYI: This should be whatever name is associated with the breakpoint number. So the category name for the category rule, and otherwise the name of the exception type.